tests/test_unit/test_filtering.py

```diff
@@ -10,115 +10,235 @@
     savgol_filter,
 )
 
+# Dataset fixtures
+list_valid_datasets_without_nans = [
+    "valid_poses_dataset",
+    "valid_bboxes_dataset",
+]
+list_valid_datasets_with_nans = [
+    f"{dataset}_with_nan" for dataset in list_valid_datasets_without_nans
+]
+list_all_valid_datasets = (
+    list_valid_datasets_without_nans + list_valid_datasets_with_nans
+)
+
+
+# Expected number of nans in the position array per individual,
+# for each dataset
+expected_n_nans_in_position_per_indiv = {
+    "valid_poses_dataset": {0: 0, 1: 0},
+    # filtering should not introduce nans if input has no nans
+    "valid_bboxes_dataset": {0: 0, 1: 0},
+    # filtering should not introduce nans if input has no nans
+    "valid_poses_dataset_with_nan": {0: 7, 1: 0},
+    # individual with index 0 has 7 frames with nans in position after
+    # filtering individual with index 1 has no nans after filtering
+    "valid_bboxes_dataset_with_nan": {0: 7, 1: 0},
+    # individual with index 0 has 7 frames with nans in position after
+    # filtering individual with index 0 has no nans after filtering
+}
+
 
 @pytest.mark.parametrize(
-    "max_gap, expected_n_nans", [(None, 0), (1, 8), (2, 0)]
+    "valid_dataset_with_nan",
+    list_valid_datasets_with_nans,
 )
-def test_interpolate_over_time(
-    valid_poses_dataset_with_nan, helpers, max_gap, expected_n_nans
+@pytest.mark.parametrize(
+    "max_gap, expected_n_nans_in_position", [(None, 0), (0, 3), (1, 2), (2, 0)]
+)
+def test_interpolate_over_time_on_position(
+    valid_dataset_with_nan,
+    max_gap,
+    expected_n_nans_in_position,
+    helpers,
+    request,
 ):
-    """Test that the number of NaNs decreases after interpolating
+    """Test that the number of NaNs decreases after linearly interpolating
     over time and that the resulting number of NaNs is as expected
     for different values of ``max_gap``.
     """
-    # First dataset with time unit in frames
-    data_in_frames = valid_poses_dataset_with_nan.position
-    # Create second dataset with time unit in seconds
-    data_in_seconds = data_in_frames.copy()
-    data_in_seconds["time"] = data_in_seconds["time"] * 0.1
-    data_interp_frames = interpolate_over_time(data_in_frames, max_gap=max_gap)
-    data_interp_seconds = interpolate_over_time(
-        data_in_seconds, max_gap=max_gap
-    )
-    n_nans_before = helpers.count_nans(data_in_frames)
-    n_nans_after_frames = helpers.count_nans(data_interp_frames)
-    n_nans_after_seconds = helpers.count_nans(data_interp_seconds)
+    valid_dataset = request.getfixturevalue(valid_dataset_with_nan)
+
+    # Get position array with time unit in frames & seconds
+    # assuming 10 fps = 0.1 s per frame
+    position = {
+        "frames": valid_dataset.position,
+        "seconds": valid_dataset.position * 0.1,
```

Comment:
I'm confused about this, you seem to be scaling the actual position pixel values here, instead of scaling the time coordinates. Shouldn't you be fiddling with the time dimesions instead of the position data variable?

Reply:
thanks for catching this, I think my brain was fried or something 😅 

Fixed [now](https://github.com/neuroinformatics-unit/movement/pull/267/commits/989c5fa628cacd9026d0dce1d7b41fd20eb998f6)